osquery/remote/transports/tests/tls_transports_tests.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <random>
+#include <thread>
+
+#include <signal.h>
+
+#include <gtest/gtest.h>
+
+#include <boost/noncopyable.hpp>
+
+#include <osquery/logger.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/serializers/json.h"
+#include "osquery/remote/transports/tls.h"
+
+#include "osquery/core/test_util.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+class TLSServerRunner : private boost::noncopyable {
```

Comment:
This is as least-jank as I could get without over engineering. I spent the weekend trying to isolate SIGSEGVs in various compile combinations (optimized, symbols included, compilers, stdc++ implementations, OS/platform version) and could not find a reliable way to start a boost::network::http::server and several clients in the same process.

A python TLS/HTTPS is simple. It self destructs after 5 seconds so a very latent set of HTTP client requests could fail and introduce confounds. There is also a second delay while we wait for the socket to set up. This could be better optimized with some wasted cycles polling `listening_ports` for the known `port_`. This server will force-tear down the python process if the set of tests/process exits before the server self destructs (at global static dtor time).